components/input/ColorPicker.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'react-components';
+import { ChromePicker } from 'react-color';
+
+const ColorPicker = ({ text, initialColor, onColorChange, ...rest }) => {
+    const [pickerState, setPickerState] = useState({
+        display: false,
+        color: initialColor
+    });
+
+    const rgbaColor = (color) => {
+        return `rgba(${color.rgb.r}, ${color.rgb.g}, ${color.rgb.b}, ${color.rgb.a})`;
+    };
+
+    const handleClick = () => {
+        setPickerState((prevState) => {
```

Comment:
Shorter version:
``` js
setPickerState({
   ...pickerState,
   display: !pickerState.display
});
```

Reply:
The shorter version that you propose does work, but I'm a bit puzzled. Since the new state is making use of the previous state, I thought one would need a [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates). That's why I fed a function to `setPickerState`... :thinking:

Reply:
I decided to split the state object into two variables since they're not really correlated. I followed the recommendations from the [React docs](https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables).
The code is cleaner and as a byproduct this issue does not exist (at any rate the recommendation by the React guide is to use the functional update as I was doing)